🐛 cosign requires authorising with device flow

Images pushed to Docker Hub but `cosign` wanted authorising

```bash
Run cosign sign --yes ministryofjustice/data-platform-python3:0.0.3
  cosign sign --yes ministryofjustice/data-platform-python3:0.0.3
  cosign sign --yes ministryofjustice/data-platform-python3:latest
  shell: /usr/bin/bash -e {0}
  env:
    name: python3
    version: 0.0.3
    CODEQL_ACTION_VERSION: [2](https://github.com/ministryofjustice/data-platform/actions/runs/4366775294/jobs/7637227745#step:18:2).2.5
    CODEQL_ACTION_FEATURE_SARIF_COMBINE: true
    CODEQL_ACTION_FEATURE_WILL_UPLOAD: true
    CODEQL_ACTION_ANALYSIS_KEY: .github/workflows/container-images.yml:build
    CODEQL_WORKFLOW_STARTED_AT: 202[3](https://github.com/ministryofjustice/data-platform/actions/runs/4366775294/jobs/7637227745#step:18:3)-03-08T17:1[4](https://github.com/ministryofjustice/data-platform/actions/runs/4366775294/jobs/7637227745#step:18:4):48.488Z
    CODEQL_UPLOAD_SARIF__GITHUB_WORKFLOWS_CONTAINER_IMAGES_YML_BUILD_IMAGES_PYTHON3__GRYPE: CODEQL_UPLOAD_SARIF__GITHUB_WORKFLOWS_CONTAINER_IMAGES_YML_BUILD_IMAGES_PYTHON3__GRYPE
    CODEQL_UPLOAD_SARIF__GITHUB_WORKFLOWS_CONTAINER_IMAGES_YML_BUILD_IMAGES_PYTHON3__TRIVY: CODEQL_UPLOAD_SARIF__GITHUB_WORKFLOWS_CONTAINER_IMAGES_YML_BUILD_IMAGES_PYTHON3__TRIVY
Generating ephemeral keys...
Retrieving signed certificate...
Non-interactive mode detected, using device flow.
Enter the verification code ZQHH-LRGV in your browser at: https://oauth2.sigstore.dev/auth/device?user_code=REDACTED
Code will be valid for 300 seconds
Error: signing [ministryofjustice/data-platform-python3:0.0.3]: getting signer: getting key from Fulcio: retrieving cert: error obtaining token: expired_token
main.go:[7](https://github.com/ministryofjustice/data-platform/actions/runs/4366775294/jobs/7637227745#step:18:7)4: error during command execution: signing [ministryofjustice/data-platform-python3:0.0.3]: getting signer: getting key from Fulcio: retrieving cert: error obtaining token: expired_token
Error: Process completed with exit code 1.
```

--- a/.github/workflows/container-images.yml
+++ b/.github/workflows/container-images.yml
@@ -79,7 +79,7 @@ jobs:
         uses: anchore/scan-action@dafbc97d7259af88b61bd260f2fde565d0668a72 # v3.3.4
         with:
           image: ${{ env.name }}:${{ env.version }}-amd64
-          # severity-cutoff: critical
+          severity-cutoff: critical
 
       - name: Upload SARIF (anchore)
         if: always()
@@ -87,7 +87,6 @@ jobs:
         uses: github/codeql-action/upload-sarif@32dc499307d133bb5085bae78498c0ac2cf762d5 # v2.2.5
         with:
           sarif_file: ${{ steps.scan_image_anchore.outputs.sarif }}
-          category: ${{ env.name }}
 
       - name: Scan SBOM
         id: scan_sbom

--- a/.github/workflows/container-images.yml
+++ b/.github/workflows/container-images.yml
@@ -42,6 +42,7 @@ jobs:
     name: Build image
     runs-on: ubuntu-latest
     permissions:
+      id-token: write
       security-events: write
     strategy:
       matrix:

--- a/.github/workflows/container-images.yml
+++ b/.github/workflows/container-images.yml
@@ -45,6 +45,7 @@ jobs:
       id-token: write
       security-events: write
     strategy:
+      fail-fast: false
       matrix:
         images: ${{ fromJson(needs.detect-changes.outputs.images) }}
     steps:
@@ -77,6 +78,7 @@ jobs:
         id: scan_image_anchore
         uses: anchore/scan-action@dafbc97d7259af88b61bd260f2fde565d0668a72 # v3.3.4
         with:
+          fail-build: false
           image: ${{ env.name }}:${{ env.version }}-amd64
 
       - name: Upload SARIF (anchore)
@@ -89,6 +91,7 @@ jobs:
         id: scan_image_trivy
         uses: aquasecurity/trivy-action@1f0aa582c8c8f5f7639610d6d38baddfea4fdcee # v0.9.2
         with:
+          exit-code: '0'
           image-ref: ${{ env.name }}:${{ env.version }}-amd64
           format: sarif
           output: trivy.sarif

--- a/.github/workflows/container-images.yml
+++ b/.github/workflows/container-images.yml
@@ -87,21 +87,6 @@ jobs:
         with:
           sarif_file: ${{ steps.scan_image_anchore.outputs.sarif }}
 
-      - name: Scan image (trivy)
-        id: scan_image_trivy
-        uses: aquasecurity/trivy-action@1f0aa582c8c8f5f7639610d6d38baddfea4fdcee # v0.9.2
-        with:
-          image-ref: ${{ env.name }}:${{ env.version }}-amd64
-          format: sarif
-          output: trivy.sarif
-
-      - name: Upload SARIF (trivy)
-        if: always()
-        id: upload_sarif_trivy
-        uses: github/codeql-action/upload-sarif@32dc499307d133bb5085bae78498c0ac2cf762d5 # v2.2.5
-        with:
-          sarif_file: trivy.sarif
-
       - name: Scan SBOM
         id: scan_sbom
         uses: anchore/scan-action@dafbc97d7259af88b61bd260f2fde565d0668a72 # v3.3.4

--- a/.github/workflows/container-images.yml
+++ b/.github/workflows/container-images.yml
@@ -78,10 +78,10 @@ jobs:
         id: scan_image_anchore
         uses: anchore/scan-action@dafbc97d7259af88b61bd260f2fde565d0668a72 # v3.3.4
         with:
-          fail-build: false
           image: ${{ env.name }}:${{ env.version }}-amd64
 
       - name: Upload SARIF (anchore)
+        if: always()
         id: upload_sarif_anchore
         uses: github/codeql-action/upload-sarif@32dc499307d133bb5085bae78498c0ac2cf762d5 # v2.2.5
         with:
@@ -91,12 +91,12 @@ jobs:
         id: scan_image_trivy
         uses: aquasecurity/trivy-action@1f0aa582c8c8f5f7639610d6d38baddfea4fdcee # v0.9.2
         with:
-          exit-code: '0'
           image-ref: ${{ env.name }}:${{ env.version }}-amd64
           format: sarif
           output: trivy.sarif
 
       - name: Upload SARIF (trivy)
+        if: always()
         id: upload_sarif_trivy
         uses: github/codeql-action/upload-sarif@32dc499307d133bb5085bae78498c0ac2cf762d5 # v2.2.5
         with:

--- a/.github/workflows/container-images.yml
+++ b/.github/workflows/container-images.yml
@@ -79,6 +79,7 @@ jobs:
         uses: anchore/scan-action@dafbc97d7259af88b61bd260f2fde565d0668a72 # v3.3.4
         with:
           image: ${{ env.name }}:${{ env.version }}-amd64
+          # severity-cutoff: critical
 
       - name: Upload SARIF (anchore)
         if: always()
@@ -86,6 +87,7 @@ jobs:
         uses: github/codeql-action/upload-sarif@32dc499307d133bb5085bae78498c0ac2cf762d5 # v2.2.5
         with:
           sarif_file: ${{ steps.scan_image_anchore.outputs.sarif }}
+          category: ${{ env.name }}
 
       - name: Scan SBOM
         id: scan_sbom

--- a/.github/workflows/path-filter/container-images.yml
+++ b/.github/workflows/path-filter/container-images.yml
@@ -2,3 +2,4 @@
 base: ./container-images/base/**
 python3: ./container-images/python3/**
 python3-dev: ./container-images/python3-dev/**
+simulated-data-producer: ./container-images/simulated-data-producer/**

--- a/.grype.yml
+++ b/.grype.yml
@@ -1,0 +1,5 @@
+---
+ignore:
+  - fix-state: not-fixed
+  - fix-state: unkwown
+  - fix-state: wont-fix

--- a/container-images/base/CHANGELOG.md
+++ b/container-images/base/CHANGELOG.md
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.4] - 2023-03-08
+
+### Added
+
+- Adding `id-token: write` to GitHub Actions workflow
+
 ## [0.0.3] - 2023-03-08
 
 ### Added

--- a/container-images/base/config.json
+++ b/container-images/base/config.json
@@ -1,4 +1,4 @@
 {
   "name": "base",
-  "version": "0.0.3"
+  "version": "0.0.4"
 }

--- a/container-images/python3-dev/CHANGELOG.md
+++ b/container-images/python3-dev/CHANGELOG.md
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.4] - 2023-03-08
+
+### Added
+
+- Adding `id-token: write` to GitHub Actions workflow
+
 ## [0.0.3] - 2023-03-08
 
 ### Added

--- a/container-images/python3-dev/CHANGELOG.md
+++ b/container-images/python3-dev/CHANGELOG.md
@@ -7,7 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [0.0.4] - 2023-03-08
+## [0.0.4] - 2023-03-09
 
 ### Added
 

--- a/container-images/python3-dev/config.json
+++ b/container-images/python3-dev/config.json
@@ -1,4 +1,4 @@
 {
   "name": "python3-dev",
-  "version": "0.0.3"
+  "version": "0.0.4"
 }

--- a/container-images/python3/CHANGELOG.md
+++ b/container-images/python3/CHANGELOG.md
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.4] - 2023-03-08
+
+### Added
+
+- Adding `id-token: write` to GitHub Actions workflow
+
 ## [0.0.3] - 2023-03-08
 
 ### Added

--- a/container-images/python3/CHANGELOG.md
+++ b/container-images/python3/CHANGELOG.md
@@ -7,7 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [0.0.4] - 2023-03-08
+## [0.0.4] - 2023-03-09
 
 ### Added
 

--- a/container-images/python3/config.json
+++ b/container-images/python3/config.json
@@ -1,4 +1,4 @@
 {
   "name": "python3",
-  "version": "0.0.3"
+  "version": "0.0.4"
 }

--- a/container-images/simulated-data-producer/CHANGELOG.md
+++ b/container-images/simulated-data-producer/CHANGELOG.md
@@ -7,24 +7,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [0.0.4] - 2023-03-09
-
-### Added
-
-- Adding `id-token: write` to GitHub Actions workflow
-
-## [0.0.3] - 2023-03-08
-
-### Added
-
-- Adding `security-events: write` to GitHub Actions workflow
-
-## [0.0.2] - 2023-03-08
-
-### Added
-
-- Pushing to Docker Hub
-
 ## [0.0.1] - 2023-03-07
 
 ### Added

--- a/container-images/simulated-data-producer/apko.yml
+++ b/container-images/simulated-data-producer/apko.yml
@@ -1,0 +1,9 @@
+---
+include: container-images/python3/apko.yml
+
+contents:
+  packages:
+    - py3-pandas
+    - py3-psycopg2
+    - py3-requests
+    - py3-sqlalchemy

--- a/container-images/simulated-data-producer/config.json
+++ b/container-images/simulated-data-producer/config.json
@@ -1,0 +1,4 @@
+{
+  "name": "simulated-data-producer",
+  "version": "0.0.1"
+}

